src/libcore/iter/mod.rs

```diff
@@ -737,6 +733,76 @@ impl<I> Iterator for StepBy<I> where I: Iterator {
     }
 }
 
+// hidden trait for specializing iterator methods
+// current use is limited to next() on StepBy
+trait SpecIter {
```

Comment:
Could this trait mention `StepBy` in its name just to make it a bit more clear that it's limited to those impls?